src/WebJobs.Script/Description/DotNet/Compilation/CSharp/ICSharpCompilation.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    public interface ICSharpCompilation : IDotNetCompilation
+    {
+        Compilation Compilation { get; }
```

Comment:
Is Compilation C# specific? I thought it was general - I.e., can't it live on the base interface?

Reply:
This is a Roslyn type, which is not used by the other .NET compilation services (i.e. F#), so we can't really move it down without introducing yet another abstraction.